src/hooks/useFingerprint.ts

```diff
@@ -1,14 +1,36 @@
 'use client';
+import FingerprintJS from '@fingerprintjs/fingerprintjs';
 import { useEffect, useState } from 'react';
 
 export const useFingerprint = () => {
-  const [fingerprint, setFingerprint] = useState<string>();
+  const [fingerprint, setFingerprint] = useState<string | undefined>(() => {
+    if (
+      typeof window !== 'undefined' &&
```

Comment:
A hook named `useSession` already exists in the codebase, better to use it please 🙏